src/overlays/actors/ovl_Mir_Ray/z_mir_ray.c

```diff
@@ -30,25 +42,537 @@ const ActorInit Mir_Ray_InitVars = {
+                }
+
+                spE8 = spE8; // Required to match
+
+                sp10C.x = (shieldMtx->xx * 100.0f) + vecB.x;
+                sp10C.y = (shieldMtx->xy * 100.0f) + vecB.y;
+                sp10C.z = (shieldMtx->xz * 100.0f) + vecB.z;
+
+                sp100.x = (spE8.x * 4.0f) + sp10C.x;
+                sp100.y = (spE8.y * 4.0f) + sp10C.y;
+                sp100.z = (spE8.z * 4.0f) + sp10C.z;
+
+                normalVec = normalVec; // Required to match
+
+                currentReflection->mtx.wz = 0.0f;
+
+                if (1) {}
+                if (1) {}
+                if (1) {}
+                if (1) {} // All four required to match
+
+                currentReflection->mtx.xx = currentReflection->mtx.yy = currentReflection->mtx.zz =
+                    currentReflection->mtx.ww = 1.0f;
+                currentReflection->mtx.xy = currentReflection->mtx.xz = currentReflection->mtx.xw =
+                    currentReflection->mtx.yx = currentReflection->mtx.yz = currentReflection->mtx.yw =
+                        currentReflection->mtx.zx = currentReflection->mtx.zy = currentReflection->mtx.zw =
+                            currentReflection->mtx.wx = currentReflection->mtx.wy = currentReflection->mtx.wz;
+
+                if (Math3D_LineSegVsPlane(normalVec.x, normalVec.y, normalVec.z,
+                                          currentReflection->reflectionPoly->dist, &sp10C, &sp100, &intersection, 1)) {
+                    currentReflection->mtx.xx = intersection.x - sp118.x;
+                    currentReflection->mtx.xy = intersection.y - sp118.y;
+                    currentReflection->mtx.xz = intersection.z - sp118.z;
+                }
+
+                sp10C.x = (shieldMtx->yx * 100.0f) + vecB.x;
+                sp10C.y = (shieldMtx->yy * 100.0f) + vecB.y;
+                sp10C.z = (shieldMtx->yz * 100.0f) + vecB.z;
+
+                sp100.x = (spE8.x * 4.0f) + sp10C.x;
+                sp100.y = (spE8.y * 4.0f) + sp10C.y;
+                sp100.z = (spE8.z * 4.0f) + sp10C.z;
+
+                if (Math3D_LineSegVsPlane(normalVec.x, normalVec.y, normalVec.z,
+                                          currentReflection->reflectionPoly->dist, &sp10C, &sp100, &intersection, 1)) {
+                    currentReflection->mtx.yx = intersection.x - sp118.x;
+                    currentReflection->mtx.yy = intersection.y - sp118.y;
+                    currentReflection->mtx.yz = intersection.z - sp118.z;
+                }
+            } else {
+                currentReflection->reflectionPoly = NULL;
+            }
+        }
+    }
+}
+
+void MirRay_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    MirRay* this = THIS;
+    Player* player = PLAYER;
+    s32 i;
+    MirRayShieldReflection reflection[6];
+    s32 temp;
+
+    this->reflectIntensity = 0.0f;
+    if ((D_80B8E670 == 0) && !(this->unLit) && Player_HasMirrorShieldSetToDraw(globalCtx)) {
+        Matrix_Mult(&player->shieldMf, 0);
```

Comment:
Any Matrix_ function that expects a `mode` argument should be called with MTXMODE_* enum values.
For instance:
```suggestion
        Matrix_Mult(&player->shieldMf, MTXMODE_NEW);
```